### DISCUSSION TITLE
Maximum number of recursions

### DISCUSSION THREAD

**Discussion by Arminsj** [Anura3D/Anura3D_OpenSource#18] [Q&A] [answered]
Hello,
I am trying to model a slope failure and want to find out the runout distance; However, after several steps, I faced an error which says that the "Maximum Number of recursions reached." 
![image](https://user-images.githubusercontent.com/105737149/177911678-c56facfd-c678-451a-a14d-4d6c77842f96.png)
Thanks

**Comment by Arminsj** [comment, marked as answer]
I found the following answer by Murphy-John on a similar problem:
"This error occurs when the code cannot find a new element after the position of a Material Point is updated. I don't have much experience with this issue but you could use the debug feature in Visual Studio to locate which Material Point is causing the error and find out why it happened.

I recall having this issue in some of my models. I think it was caused by a boundary condition not applied correctly, so it may be worth reviewing your boundary conditions."
Also, Francesca Ceccato added:
"This error means that the algorithm could not find an element for a moving material point, maybe it is moving too fast (instability issue) or boundary conditions are not correctly applied or there is an issue with the geometry."

**Reply by FrancescaCeccato** [reply]
Yes exactly, the problem must be somewhere else... check your output and maybe you get and idea to solve the issue!

**Comment by GeoCalcul** [comment]
Hi,
I am getting the same error when I use a finer mesh size for a tutorial example, also more material points.
